scripts/zones/Riverne-Site_B01/mobs/Bahamut.lua

```diff
@@ -1,15 +1,97 @@
 -----------------------------------
 -- Area: Riverne - Site B01 (BCNM)
--- NPC:  Bahamut
+--  NM:  Bahamut
 -----------------------------------
 
-require("scripts/globals/titles");
+require("scripts/globals/quests");
+require("scripts/globals/status");
 
 -----------------------------------
 -- onMobSpawn Action
 -----------------------------------
 
 function onMobSpawn(mob)
+    mob:addStatusEffect(EFFECT_PHALANX,35,0,180);
+    mob:addStatusEffect(EFFECT_STONESKIN,350,0,300);
+    mob:addStatusEffect(EFFECT_PROTECT,175,0,1800);
+    mob:addStatusEffect(EFFECT_SHELL,62,0,1800);
+end;
+
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
+
+function onMobFight(mob,target)
+    local MegaFlareQueue = mob:getLocalVar("MegaFlareQueue");
+    local MegaFlareTrigger = mob:getLocalVar("MegaFlareTrigger");
+    local MegaFlareUses = mob:getLocalVar("MegaFlareUses");
+    local FlareWait = mob:getLocalVar("FlareWait")
+    local GigaFlare = mob:getLocalVar("GigaFlare");
+    local tauntShown = mob:getLocalVar("tauntShown");
+    local mobHPP = mob:getHPP();
+    local isBusy = false;
+    if (GetMobAction(mob:getID()) == ACTION_MOBABILITY_START or GetMobAction(mob:getID()) == ACTION_MOBABILITY_USING or GetMobAction(mob:getID()) == ACTION_MOBABILITY_FINISH or GetMobAction(mob:getID()) == ACTION_MAGIC_START or GetMobAction(mob:getID()) == ACTION_MAGIC_CASTING or GetMobAction(mob:getID()) == ACTION_MAGIC_START) then

```

Comment:
the queue is empty as soon as he starts using the move. it would only be useful here if the conditions for mega/gigaflare were changed when the move was started.  jormungand/ouryu/tiamat use it to make sure they don't touchdown multiple times in a row if the time ticked while casting a spell